src/data/base-media-service.js

```diff
@@ -0,0 +1,15 @@
+import decodeMediaData from '~/utils/decode-media-data.js'
+
+const BaseMediaService = (mediaType) => ({
+  transformResults(data) {
+    data.results = Object.fromEntries(
+      data.results.map((item) => {
+        item = decodeMediaData(item, mediaType)
+        return [item.id, item]
+      })
+    )
```

Comment:
Couldn't this be done in a single pass using `reduce`? I'm not sure how big `data.results` can get but it could be significant looping `n`, allocating `n` arrays then throwing them all out to do another `n` loop.

```suggestion
    data.results = data.results.reduce((acc, item) => {
      acc[item.id] = decodeMediaData(item, mediaType)
      return acc
    }, {})
```

Reply:
That's a great idea!